frontend/packages/console-dynamic-plugin-sdk/src/api/dynamic-core-api.ts

```diff
@@ -883,6 +884,26 @@ export const useLabelsModal: UseLabelsModal = require('@console/shared/src/hooks
 export const useActiveNamespace: UseActiveNamespace = require('@console/shared/src/hooks/useActiveNamespace')
   .useActiveNamespace;
 
+/**
+ * A hook that provides a callback to launch a modal for open the CreateProjectModal or CreateNamespaceModal.
```

Comment:
I haven't dug into the code, but it's a little odd that it will launch one or the other. Do we know when one or the other is chosen? Is it based on RBAC or something else? We should add that to the description.

Is there any reason we can't expose a hook that only launches the create project modal instead, or does your plugin really need the namespace modal in some cases?

Reply:
Yes it's based on the flag `const isOpenShift = useFlag(FLAGS.OPENSHIFT);`

https://github.com/upalatucci/console/blob/master/frontend/packages/console-shared/src/hooks/useCreateNamespaceOrProjectModal.ts

Reply:
We also have `useCreateProjectModal` and `useCreateNamespaceModal` but I'm not sure what we want to expose

Reply:
OK, we should explain that in the description.